### ISSUE TITLE
Clean up new netcore50 code in System.Net.Http, and reconcile with existing code

### ISSUE THREAD

**Issue by mellinoe** [dotnet/corefx#5525]
See the list of files in the System.Net.Http project file under the netcore50 TargetGroup:

``` XML
<ItemGroup Condition="'$(TargetGroup)' == 'netcore50'">
    <Compile Include="$(CommonPath)\System\NotImplemented.cs">
      <Link>Common\System\NotImplemented.cs</Link>
    </Compile>
    <!-- TODO: reconcile with Open\src\Common, see issue #REPLACE_ME -->
    <Compile Include="$(CommonPath)\System\Net\HttpVersion.cs">
      <Link>Common\System\Net\HttpVersion.cs</Link>
    </Compile>
      <Compile Include="$(CommonPath)\System\StringExtensions.cs">
      <Link>Common\System\StringExtensions.cs</Link>
    </Compile>
    <Compile Include="netcore50\System\Net\cookie.cs" />
    <Compile Include="netcore50\System\Net\cookieexception.cs" />
    <Compile Include="netcore50\System\Net\CookieHelper.cs" />
    <Compile Include="netcore50\System\Net\HttpHandlerToFilter.cs" />
    <Compile Include="netcore50\System\Net\HttpClientHandler.cs" />
    <Compile Include="netcore50\System\Net\RtcRequestFactory.cs" />
  </ItemGroup>
```

Additionally, the code could use a bit of cleanup in general. Things like field names, brace usage, comment formatting, etc.

**Comment by karelz** [comment]
This should be actionable once we have UWP testing back in place.

**Comment by CIPop** [comment]
@davidsh I remember a PR where you were changing this. Is it now fixed?

**Comment by davidsh** [comment]
Fixed with #21905
